### PR TITLE
osd/osd-rep-recov-eio.sh: TEST_rados_repair_warning:  return 1

### PR DESCRIPTION
**Pull request by dzafman**
## Checklist
- [x] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Comment by dzafman** [comment]
Testing passed with other failures
https://pulpito.ceph.com/dzafman-2020-09-29_20:13:01-rados-wip-zafman-testing-46405-distro-basic-smithi/

File: qa/standalone/osd/osd-rep-recov-eio.sh [hunk 1]
```diff
@@ -120,6 +120,7 @@ function rados_get_data() {
     inject_$inject rep data $poolname $objname $dir 0 || return 1
     rados_get $dir $poolname $objname || return 1
 
+    sleep 1
```
Comment (badone) [inline qa/standalone/osd/osd-rep-recov-eio.sh:123]: Is there any way to poll whether the repair is finished?

**Review by badone — CHANGES_REQUESTED** [review]

File: qa/standalone/osd/osd-rep-recov-eio.sh [hunk 2]
```diff
@@ -224,6 +228,7 @@ function TEST_rados_repair_warning() {
        rados_get $dir $poolname ${objbase}-$i || return 1
     done
 
+    wait_for_clean
```
[the remark this answers is not in the document]
Comment (dzafman) [reply qa/standalone/osd/osd-rep-recov-eio.sh:231]: After a read that needs a repair, recovery is started.  I never noticed on my build machine that the query could happen before the recovery is done, but teuthology did have this race.  I did a sleep then Brad asked if there was some other way to tell of the repair is finished.  So I realized that wait_for_clean() would do it.  The only other possible race, is if recovery was async to the read repair.  So if after the read the PG was still active+clean we would have a problem.

[the remark this answers is not in the document]
Comment (neha-ojha) [reply qa/standalone/osd/osd-rep-recov-eio.sh:231]: I've just seen the test fail here so wondering whether we need to add `wait_for_clean` in all the other places or not, certainly doesn't hurt

File: qa/standalone/osd/osd-rep-recov-eio.sh (hunk 1 above)
Comment (dzafman) [reply qa/standalone/osd/osd-rep-recov-eio.sh]: Switched to wait_for_clean().

**Review by badone — APPROVED** [review]

**Review by neha-ojha — APPROVED** [review]

**Comment by neha-ojha** [comment]
@dzafman is this ready to merge?

**Comment by dzafman** [comment]
@neha-ojha Yes

**Pull request merged**